plugins/polio/api/vaccines/stock_management.py

```diff
@@ -767,33 +784,64 @@ def create(self, request, *args, **kwargs):
 
 class IncidentReportSerializer(serializers.ModelSerializer):
     document = serializers.FileField(required=False)
+    can_edit = serializers.SerializerMethodField()
 
     class Meta:
         model = IncidentReport
         fields = "__all__"
 
+    def get_can_edit(self, obj):
+        return can_edit_helper(
+            self.context["request"].user,
+            obj.created_at,
+            admin_perm=permission.POLIO_VACCINE_STOCK_MANAGEMENT_WRITE,
+            non_admin_perm=permission.POLIO_VACCINE_STOCK_MANAGEMENT_READ,
+        )
+
 
 class IncidentReportViewSet(VaccineStockSubitemBase):
     serializer_class = IncidentReportSerializer
     model_class = IncidentReport
+    permission_classes = [
+        lambda: VaccineStockManagementPermission(
+            non_admin_perm=permission.POLIO_VACCINE_STOCK_MANAGEMENT_READ,
+            admin_perm=permission.POLIO_VACCINE_STOCK_MANAGEMENT_WRITE,
+        )
+    ]
 
 
 class DestructionReportSerializer(serializers.ModelSerializer):
     document = serializers.FileField(required=False)
+    can_edit = serializers.SerializerMethodField()
 
     class Meta:
         model = DestructionReport
         fields = "__all__"
 
+    def get_can_edit(self, obj):
+        return can_edit_helper(
+            self.context["request"].user,
+            obj.created_at,
+            admin_perm=permission.POLIO_VACCINE_STOCK_MANAGEMENT_WRITE,
+            non_admin_perm=permission.POLIO_VACCINE_STOCK_MANAGEMENT_READ,
+        )
+
 
 class DestructionReportViewSet(VaccineStockSubitemBase):
     serializer_class = DestructionReportSerializer
     model_class = DestructionReport
+    permission_classes = [
+        lambda: VaccineStockManagementPermission(
+            non_admin_perm=permission.POLIO_VACCINE_STOCK_MANAGEMENT_READ,
+            admin_perm=permission.POLIO_VACCINE_STOCK_MANAGEMENT_WRITE,
```

Comment:
It's confusing that you put them in "non-admin, admin" order where in the constructor you did "admin, non-admin."

This should be consistent